libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -0,0 +1,497 @@
+                    node->setSoftwareVersion(sw_version);
+
+                    uavcan::protocol::HardwareVersion hw_version; // Standard type uavcan.protocol.HardwareVersion
+
+                    hw_version.major = 2;
+                    hw_version.minor = 0;
+                    node->setHardwareVersion(hw_version);
+
+                    const int node_start_res = node->start();
+                    if (node_start_res < 0) {
+                        printf("UAVCAN: node start problem\n\r");
+                    }
+
+                    gnss_fix = new uavcan::Subscriber<uavcan::equipment::gnss::Fix>(*node);
+                    const int gnss_fix_start_res = gnss_fix->start(gnss_fix_cb);
+                    if (gnss_fix_start_res < 0) {
+                        return false;
+                    }
+
+                    gnss_aux = new uavcan::Subscriber<uavcan::equipment::gnss::Auxiliary>(*node);
+                    const int gnss_aux_start_res = gnss_aux->start(gnss_aux_cb);
+                    if (gnss_aux_start_res < 0) {
+                        return false;
+                    }
+
+                    magnetic = new uavcan::Subscriber<uavcan::equipment::ahrs::MagneticFieldStrength>(*node);
+                    const int magnetic_start_res = magnetic->start(magnetic_cb);
+                    if (magnetic_start_res < 0) {
+                        return false;
+                    }
+
+                    air_data_sp = new uavcan::Subscriber<uavcan::equipment::air_data::StaticPressure>(*node);
+                    const int air_data_sp_start_res = air_data_sp->start(air_data_sp_cb);
+                    if (air_data_sp_start_res < 0) {
+                        return false;
+                    }
+
+                    air_data_st = new uavcan::Subscriber<uavcan::equipment::air_data::StaticTemperature>(*node);
+                    const int air_data_st_start_res = air_data_st->start(air_data_st_cb);
+                    if (air_data_st_start_res < 0) {
+                        return false;
+                    }
+
+                    act_out_array = new uavcan::Publisher<uavcan::equipment::actuator::ArrayCommand>(*node);
+                    act_out_array->setTxTimeout(uavcan::MonotonicDuration::fromMSec(20));
+                    act_out_array->setPriority(uavcan::TransferPriority::OneLowerThanHighest);
+
+                    esc_raw = new uavcan::Publisher<uavcan::equipment::esc::RawCommand>(*node);
+                    esc_raw->setTxTimeout(uavcan::MonotonicDuration::fromMSec(20));
+                    esc_raw->setPriority(uavcan::TransferPriority::OneLowerThanHighest);
+
+                    /*
+                     * Informing other nodes that we're ready to work.
+                     * Default mode is INITIALIZING.
+                     */
+                    node->setModeOperational();
+
+                    _initialized = true;
+
+                    printf("UAVCAN: init done\n\r");
```

Comment:
It gives output to debug console. If needs to be changed to something - I will, just point in right direction.